src/App.tsx

```diff
@@ -1,99 +1,106 @@
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+  const ref = React.useRef(null);
 
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  useOnClickOutside(ref, () => setIsOpen(false));
 
-          <div
-            className="dropdown-menu"
-            role="menu"
-            data-cy="suggestions-list"
-          >
-            <div className="dropdown-content">
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  const { name, born, died } = selectedPerson || {};
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+  const filteredPeople = useMemo(() => {
+    if (inputValueDeb) {
+      return peopleFromServer.filter(
+        person => person.name.toLocaleLowerCase().includes(
+          inputValueDeb.toLocaleLowerCase(),
+        ),
+      );
+    }
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+    return peopleFromServer;
+  }, [inputValueDeb]);
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  const debounced = useDebouncedCallback(
+    (value) => {
+      setInputValueDeb(value);
+      setIsOpen(true);
+    },
+    500,
+  );
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  const handleSelectPerson = (person:Person) => {
+    setSelectedPerson(person);
+    setInputValue(person.name);
+    setIsOpen(false);
+  };
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
+  const handleChangeText = (e:ChangeEvent<HTMLInputElement>) => {
+    setIsOpen(false);
+    setInputValue(e.target.value);
+    debounced(e.target.value);
+  };
+
+  return (
+    <main className="section">
+      <h1 className="title">
+        {selectedPerson ? `${name} (${born} = ${died})` : 'No selected person'}
```

Comment:
Move this logic to the variable and use it here
```suggestion
        {selectedPerson ? `${name} (${born} - ${died})` : 'No selected person'}
```